### ISSUE TITLE
Adding Ruby parser

### ISSUE THREAD

**Issue by peter-leonov** [fkling/astexplorer#302]
Hi guys,

Thanks for the inspiring and useful project! Learning babel transformations with it is a pure joy 👍

Are you interested in adding a Ruby parser? I've developed [ruby-parser.js](http://peter-leonov.github.io/ruby-parser.js/) for fun in pure JavaScript. It can also be of use to someone learning Ruby. I'm fine with making a PR if you find it interesting.

**Comment by fkling** [comment]
In general yes, though the output of the parser doesn't seem to be suited very well for astexplorer. Nodes are usually objects with types.

**Comment by peter-leonov** [comment]
IMHO, converting S-expressions to node is a quite easy task. Are there any other requirements you see as a must?

**Comment by fkling** [comment]
One thing that people find most useful is the fact that we can highlight the corresponding source text of a node and "focus" the corresponding code when moving the cursor around in the source text.

To do this we need to know a node's start and end index in the source text string.

**Comment by peter-leonov** [comment]
True, I find it awesome too :) As far as I remember the AST part of my ruby parser lacks location information, but the lexer does store token locations. So it wouldn't be that hard to make AST location aware also. Will check it out.

Thanks for a quick and informative reply!
